lib/preset-patterns.js

```diff
@@ -18,10 +18,10 @@ module.exports = {
  */
 function suitSelector(componentName, presetOptions) {
   var ns = (presetOptions && presetOptions.namespace) ? presetOptions.namespace + '-' : '';
-  var OPTIONAL_PART = '(?:-[a-zA-Z0-9]+)?';
-  var OPTIONAL_MODIFIER = '(?:--[a-zA-Z0-9]+)?';
+  var OPTIONAL_PART = '(?:-[a-z][a-zA-Z0-9]*)?';
+  var OPTIONAL_MODIFIER = '(?:--[a-z0-9][a-zA-Z0-9]*)?';
   var OPTIONAL_ATTRIBUTE = '(?:\\[\\S+\\])?';
-  var OPTIONAL_STATE = '(?:\\.is-[a-zA-Z0-9]+)?';
+  var OPTIONAL_STATE = '(?:\\.is-[a-z][a-zA-Z0-9]*)?';
```

Comment:
Sorry, just noticed this ... @simonsmith, should descendent parts and states accept initial numbers as well as lowercase letters, like modifier? E.g. are these classes legit: `.Clothes-2pieceSuit`, `.is-4sale`?